docs/docs/sample-scripts/teams/export-teams-conversations.md

```diff
@@ -0,0 +1,95 @@
+# Export conversations from Microsoft Teams Channels
+
+Author: [Joseph Velliah](https://sprider.blog/export-conversations-from-microsoft-teams)
+
+## Problem Statement
+
+We utilize Teams during incidents and create channels for each. We would like to be able to export conversation history.
+
+- Right now the only option we have is to go through Security & Compliance.
+- Teams usage is growing in every organization and it would soon become unreasonably to only have Administrators be the ones doing exports of channels for all the Teams.
+
+### Solution
+
+This script uses Microsoft 365 CLI to export the conversations from Microsoft Teams Channels.
+
+!!! notes
+
+- Commands m365 teams message list and reply list are based on an API that is currently in preview and is subject to change once the API reached general availability.
+- You can only retrieve a message from a Microsoft Teams team if you are a member of that team.
+
+```powershell tab="PowerShell Core"
+function  Get-Teams {
+    $teams = m365 teams team list -o json | ConvertFrom-Json -AsHashtable
+    return $teams
+}
+function  Get-Channels {
+    param (
+        [Parameter(Mandatory = $true)] [string] $teamId
+    )
+    $channels = m365 teams channel list --teamId $teamId -o json | ConvertFrom-Json -AsHashtable
+    return $channels
+}
+function  Get-Messages {
+    param (
+        [Parameter(Mandatory = $true)] [string] $teamId,
+        [Parameter(Mandatory = $true)] [string] $channelId
+    )
+    $messages = m365 teams message list --teamId $teamId --channelId $channelId -o json | ConvertFrom-Json -AsHashtable
+    return $messages
+}
+function  Get-MessageReplies {
+    param (
+        [Parameter(Mandatory = $true)] [string] $teamId,
+        [Parameter(Mandatory = $true)] [string] $channelId,
+        [Parameter(Mandatory = $true)] [string] $messageId
+    )
+
+    $messageReplies = m365 teams message reply list --teamId $teamId --channelId $channelId --messageId $messageId -o json | ConvertFrom-Json -AsHashtable
+    return $messageReplies
+}
+
+$teamsCollection = [System.Collections.ArrayList]@()
+$teams = Get-Teams
+$teamsCount = $teams.Length
+Write-Host "$teamsCount Team/s found" -ForegroundColor Magenta
```

Comment:
Could we use PowerShell's progress? I think it's more natural for people using PowerShell and it's easier to keep track how for in the script currently is, especially if you have many Teams and Channels and the output spans the height of the terminal.

Reply:
Sure, I have added progress